### PR TITLE
Handles token related errors

### PR DESCRIPTION
**Pull request by big-kahuna-burger**
Handles all jwt related errors with 401.

🗝

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,8 +25,16 @@ var Service = module.exports = function (opts) {
...
       var token = authHeader.slice(7)
       decode(token, function (err, authData) {
-        if (err && err.name === 'TokenExpiredError') err.statusCode = 401
-        cb(err, authData)
+        if (err) {
+          if (err.name === 'TokenExpiredError' ||
+            err.name === 'JsonWebTokenError') {
+            err.statusCode = 401
+            err.stack = undefined
+            return cb(null, err)
```
Comment (davidguttman) [inline index.js]: This should have the same behavior as before: it should `cb(err)`.

File: package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "authentic-service",
-  "version": "0.2.2",
+  "version": "0.3.0",
```
Comment (davidguttman) [inline package.json]: Don't bump the version.

File: package.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
   },
   "devDependencies": {
     "servertest": "^1.2.1",
+    "standard": "^10.0.2",
```
Comment (davidguttman) [inline package.json]: This PR should only do a single thing -- adding standard should be in a separate PR.

File: index.js [hunk 4]
```diff
@@ -3,19 +3,25 @@ var Corsify = require('corsify')
...
+  'JsonWebTokenError': 401,
+  'NotBeforeError': 401
+}
+
 var cors = Corsify({
   'Access-Control-Allow-Headers': 'authorization, accept, content-type'
 })
 
-var Service = module.exports = function (opts) {
+module.exports = function (opts) {
```
Comment (davidguttman) [inline index.js]: Not related to this PR

File: index.js [hunk 5]
```diff
@@ -3,19 +3,25 @@ var Corsify = require('corsify')
...
   var prefix = opts.prefix || '/auth'
   var pubKeyUrl = opts.server + prefix + '/public-key'
   var cache = createCache(pubKeyUrl, opts.cacheDuration)
 
   function decode (token, cb) {
     cache.get('pubKey', function (err, pubKey) {
       if (err) return cb(err)
-      jwt.verify(token, pubKey, {algorithms: ['RS256']}, cb)
+      jwt.verify(token, pubKey, { algorithms: ['RS256'] }, cb)
```
Comment (davidguttman) [inline index.js]: Not related to this PR

File: index.js [hunk 6]
```diff
@@ -42,7 +48,7 @@ function createCache (pubKeyUrl, cacheDuration) {
       jsonist.get(pubKeyUrl, function (err, body) {
         if (err) return cb(err)
 
-        var pubKey = ((body||{}).data||{}).publicKey
+        var pubKey = ((body || {}).data || {}).publicKey
```
Comment (davidguttman) [inline index.js]: Not related to this PR

File: test/index.js [hunk 7]
```diff
@@ -3,24 +3,23 @@ var http = require('http')
 var tape = require('tape')
 var servertest = require('servertest')
 var jwt = require('jsonwebtoken')
+var ms = require('ms')
+var join = require('path').join
 
-var publicKey = fs.readFileSync(__dirname + '/rsa-public.pem', 'utf-8')
-var privateKey = fs.readFileSync(__dirname + '/rsa-private.pem', 'utf-8')
+var publicKey = fs.readFileSync(join(__dirname, 'rsa-public.pem'), 'utf-8')
+var privateKey = fs.readFileSync(join(__dirname, 'rsa-private.pem'), 'utf-8')
```
Comment (davidguttman) [inline test/index.js]: Not related to this PR

File: test/index.js [hunk 8]
```diff
@@ -3,24 +3,23 @@ var http = require('http')
...
-var token = jwt.sign(payload, privateKey, {algorithm: 'RS256'})
-
 tape('init', function (t) {
   server = http.createServer(function (req, res) {
     if (req.url !== '/auth/public-key') return
     res.end(JSON.stringify({
-      "success": true,
-      "data": { "publicKey": publicKey }
+      'success': true,
+      'data': { 'publicKey': publicKey }
```
Comment (davidguttman) [inline test/index.js]: Not related to this PR (and you can remove the quotes around the keys later)

File: test/index.js [hunk 9]
```diff
@@ -34,27 +33,125 @@ tape('init', function (t) {
 })
 
 tape('should handle anonymous request', function (t) {
-  var opts = {method: 'GET'}
+  var opts = { method: 'GET' }
   servertest(createService(auth), '/', opts, function (err, res) {
     t.ifError(err, 'should not error')
+
     var data = JSON.parse(res.body)
```
Comment (davidguttman) [inline test/index.js]: Not related to this PR

File: test/index.js [hunk 10]
```diff
@@ -34,27 +33,125 @@ tape('init', function (t) {
...
 
   servertest(createService(auth), '/', opts, function (err, res) {
     t.ifError(err, 'should not error')
     var data = JSON.parse(res.body)
 
     t.equal(data.email, 'chet@scalehaus.io', 'should have correct email')
-    t.equal(data.expiresIn, '30d', 'should have correct expiresIn')
-    t.ok(data.iat, 'chet@scalehaus.io', 'should have iat')
+    t.equal(data.exp, ~~(expiresInMs / 1000), 'should have correct exp claim')
+    t.equal(data.iat, ~~(now / 1000), 'should have correct iat')
```
Comment (davidguttman) [inline test/index.js]: Why is this test being changed?

Comment (big-kahuna-burger) [inline test/index.js]: Let me respond with question to this:
line 56: should `expiresIn` be just data property? If so, than test should not be changed. 

Docs say that when you sign with expiresIn in signOpts, than payload gets a 'exp' claim, and not `expiresIn`

ln: 57 also, `should have iat` string is shown nowhere in console when I run tests, and tape's `ok` func has params signature as `ok(value: any, msg?: string): void;`

Comment (davidguttman) [inline test/index.js]: It's been awhile since I've worked with this, so I don't know off-hand about `expiresIn`, but a valid test shouldn't be changed without good reason.

L57 is probably a copy-paste artifact, the second/email argument is a mistake.
